src/components/ha-select-box.ts

```diff
@@ -0,0 +1,193 @@
+import { customElement, property } from "lit/decorators";
+import { css, html, LitElement, nothing } from "lit";
+import "./ha-radio";
+import { classMap } from "lit/directives/class-map";
+import { styleMap } from "lit/directives/style-map";
+import type { HaRadio } from "./ha-radio";
+import { fireEvent } from "../common/dom/fire_event";
+
+export interface SelectBoxOption {
+  label?: string;
+  description?: string;
+  image?: string;
+  value: string;
+  disabled?: boolean;
+}
+
+@customElement("ha-select-box")
+export class HaSelectBox extends LitElement {
+  @property({ attribute: false }) public options: SelectBoxOption[] = [];
+
+  @property({ attribute: false }) public value?: string;
+
+  @property({ type: Boolean }) public disabled?: boolean;
+
+  @property({ type: Number, attribute: "max_columns" })
+  public maxColumns?: number;
+
+  render() {
+    const maxColumns = this.maxColumns ?? 3;
+    const columns = Math.min(maxColumns, this.options.length);
+
+    return html`
+      <div class="list" style=${styleMap({ "--columns": columns })}>
+        ${this.options.map((option) => this._renderOption(option))}
+      </div>
+    `;
+  }
+
+  private _renderOption(option: SelectBoxOption) {
+    const horizontal = this.maxColumns === 1;
+    const disabled = option.disabled || this.disabled || false;
+    const selected = option.value === this.value;
+    return html`
+      <label
+        class="option ${classMap({
+          horizontal: horizontal,
+          selected: selected,
+        })}"
+        ?disabled=${disabled}
+        @click=${this._labelClick}
+      >
+        <div class="content">
+          <ha-radio
+            .checked=${option.value === this.value}
+            .value=${option.value}
+            .disabled=${disabled}
+            @change=${this._radioChanged}
+          ></ha-radio>
+          <div class="text">
+            <span class="label">${option.label}</span>
+            ${option.description
+              ? html`<span class="description">${option.description}</span>`
+              : nothing}
+          </div>
+        </div>
+        ${option.image ? html`<img alt="" src=${option.image} />` : nothing}
```

Comment:
We should provide an `option.imageDescription` to fill the alt property.

Reply:
I'm not sure it's necessary. I think we should add aria hidden for this one as it's a really useful image for screen reader.